pdr_backend/lake/fetch_ohlcv.py

```diff
@@ -114,3 +158,136 @@ def _filter_within_timerange(
 ) -> list:
     uts = _ohlcv_to_uts(tohlcv_data)
     return [vec for ut, vec in zip(uts, tohlcv_data) if st_ut <= ut <= fin_ut]
+
+
+def fetch_dydx_data(
+    symbol: str, resolution: str, st_ut: UnixTimeMs, fin_ut: UnixTimeMs, limit: int
```

Comment:
Key comment: once you remove the complexity of no loop, and the conversion is done by a constant or smash separate method, the remaining work that this method does is small. So you can move all of its work into safe_fetch_ohlcv_dydx().